src/plugins/data/common/search/aggs/agg_configs.ts

```diff
@@ -78,29 +79,27 @@ export type GenericBucket = estypes.AggregationsBuckets<any> & {
 export type IAggConfigs = AggConfigs;
 
 export class AggConfigs {
-  public indexPattern: IndexPattern;
   public timeRange?: TimeRange;
   public timeFields?: string[];
   public forceNow?: Date;
-  public hierarchical?: boolean = false;
-
-  private readonly typesRegistry: AggTypesRegistryStart;
-
-  aggs: IAggConfig[];
+  public aggs: IAggConfig[] = [];
+  public hierarchical?: boolean;
+  public readonly timeZone: string;
 
   constructor(
-    indexPattern: IndexPattern,
+    public indexPattern: IndexPattern,
     configStates: CreateAggConfigParams[] = [],
-    opts: AggConfigsOptions
+    private opts: AggConfigsOptions,
+    private getConfig: GetConfigFn
   ) {
-    this.typesRegistry = opts.typesRegistry;
```

Comment:
Why this change?

Reply:
I've checked and found that we don't use it. I found an error in the clone method (the `hierarchical` property was not passed there), I decided that we should not face a similar problem again and remove this personal property